### PR TITLE
Fix #1585. I fixed the underwear. Probably. Maybe. Once and for all, if it works.

### LINKED ISSUE #1585: Undershirts are seemingly broken.
You can pick your undershirt slot and save it in your loadout no problem but A) it doesn't show up on the preview, and B) once you're actually in-game, it doesn't show up on your sprite, nor does it seem to appear as an object in the underwear removal/equipping menus.

Socks may or may not be suffering the same issue.

### PR DESCRIPTION
**Pull request by timidvi**
## About The Pull Request

Fixes: various underwear clothings in loadout which had lacked sprites, including but not limited to most of the feminine undershirts, the female sports bra, polo shirts, bowling shirts, most of the alternate versions of undershirts, and so forth.
Adds: Some undershirts which had not previously been in the code but were in the spritesheet.

To potentially do in the future: 
1. Make more of these items colourable. I'm looking at you, fishnets. Also maybe adding those army shirts and tank tops now included in the .dmi.
2.  Change the .dmi file being referenced in undershirt.dm, top.dm, bottom.dm, socks.dm, etc so that this is modular and doesn't potentially get broken by upstreams (eg. mob.dmi --> mob_eclipse.dmi). Currently unsure of exactly where the .dmi reference actually is - it does not appear to be a part of any of those, nor of the base.dm file under code/modules/clothing/underwear, else I would have fixed it myself immediately.
3. Originally pull request would have also fixed a lot of the hairs. I'll do that in the next pull request, this one's already crowded by my inexperienced commits and attempts at roll backs.

Fixes #1585 

## Why It's Good For The Game

Fixes the broken as fuck underwear system we've had for literal years. MCI fixed a few of the items but she never added the sprites in for a lot of the others... Or they just got immediately overwritten by an upstream, not sure. May or may not suffer the same problem, but this fixes it at least until the next time someone upstream messes with the underwear icons file - and that means one less immediately glaring issue that newcomers will notice while character creating.

Also, more character customization in HRP environments --> very very good at engaging players, makes people happy, gives me the happy chemicals and probably others too :3 :D .

## Changelog
:cl:
add: Added entirely missing undershirts
fix: fixed all the broken underwear items which previously were missing sprites
imageadd: added icon_states to mob.dmi 
code: slight updates to undershirt.dm to facilitate all the above
/:cl:

### CODE REVIEW

**Comment by timidvi** [comment]
https://github.com/Eclipse-Station/NEV-Northern-Light/issues/1585

**Comment by timidvi** [comment]
Additionally: Altered hair.dmi to fix hair with broken tinting in character creator.

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
Sprite changes must be made in modular; if we pull from upstream, it will be overwritten otherwise.

**Comment by timidvi** [comment]
> Sprite changes must be made in modular; if we pull from upstream, it will be overwritten otherwise.

Still not sure how to safely do that for either the hair or the underwear, especially since both aren't items per se and rather are linked to the character creator.

For the underwear, the problem is still that I don't know where it's actually referencing the filepath. It doesn't seem to be in the datums which list off the icon_states, but not the actual icon files (ie undershirt.dm, top.dm, bottom.dm, socks.dm, etc).

For the hair, the problem is that there seems to be a single line that sets all hair icon_states to reference the hair.dmi file that currently exists. Is there a way to overwrite those individual hairs to reference a different icon/dmi file?

**Comment by EvilJackCarver** [comment]
Unless it's explicitly defined otherwise, everything inherits from the parent object type. For example:

```C
/obj/foo
	name = "Blah"
	desc = "sample text"

/obj/foo/bar
	name = "Bill Nye the Science Guy"
```

Despite `/obj/foo/bar` being a different typepath, it will still have the description text of "sample text", as that was defined in `/obj/foo`. The same technique works for all variables defined.

To use a real, in-game example, in [modular /radio_sprites/radios.dm](https://github.com/Eclipse-Station/NEV-Northern-Light/blob/master/zzz_modular_eclipse/radio_sprites/radios.dm), I changed the icon files of the radio items (/obj/item/device/radio) to use ./xpr.dmi. However, due to inheritance, things like beacons (obj/item/device/radio/beacon) and headsets (/obj/item/device/radio/headset) would also point to that file, as it is not defined inside their code otherwise, so at the end of radios.dm, I had to set everything that inherited to point back towards the main radio DMI file.

The same method can be used for underwear and hairstyles, as long as the icon states are correctly set up.

**Comment by timidvi** [comment]
> 

Ohhhh okay I think I see now. I'll try setting up a modular file and you can tell me if it'd instead horribly ruin everything :rofl:

**Comment by timidvi** [comment]
Settled on making the spritesheet/.dmi file fully modular, and making dirty edits tagged with the Eclipse Edit comment around edited sections. Tried making it fully modular but it's not possible for someone like me who doesn't really know code and is mostly trying to copy logic.

tl;dr of changes:
Added modular icon file to fix underwear along file path zzz_modular_eclipse/icons/inventory/underwear/mob.dmi

Changed strings referencing icons/inventory/underwear/mob.dmi to reference zzz_modular_eclipse/icons/inventory/underwear/mob.dmi in the following files:

1. code/modules/client/preferences.dm
2. code/modules/mob/living/carbon/human/update_icons.dm
3. code/game/objects/structures/under_wardrobe.dm

Added and restored missing or commented out undershirt items in code/datums/underwear/undershirt.dm

Everything else should be functionally identical to the main repo.

**Review by EvilJackCarver — APPROVED** [review]
